src/rgw/driver/rados/rgw_d3n_datacache.cc

```diff
@@ -25,12 +25,18 @@ namespace efs = std::experimental::filesystem;
 
 using namespace std;
 
-int D3nCacheAioWriteRequest::d3n_libaio_prepare_write_op(bufferlist& bl, unsigned int len, string oid, string cache_location)
+int D3nCacheAioWriteRequest::d3n_libaio_prepare_write_op(bufferlist& bl, unsigned int len, string digest_oid, string cache_location)
 {
-  std::string location = cache_location + url_encode(oid, true);
+  std::string location = cache_location + digest_oid;
   int r = 0;
 
   lsubdout(g_ceph_context, rgw_datacache, 20) << "D3nDataCache: " << __func__ << "(): Write To Cache, location=" << location << dendl;
+  std::filesystem::path dir = std::filesystem::path(location).parent_path();
+  std::error_code sec;
+  if (!std::filesystem::exists(dir) && !std::filesystem::create_directories(dir, sec)) {
```

Comment:
is this stuff still necessary? same below in d3n_io_write()

Reply:
thank you, removed in all places